libs/core/src/lib/utils/services/carousel.service.ts

```diff
@@ -0,0 +1,297 @@
+        }
+
+        const size: number = this._getSize(this.items.first);
+
+        /** When scroll is after half of element, then ext one should be active */
+        const halfApproached: boolean = Math.abs(this._currentTransitionPx % size) > size / 2;
+        /**
+         * Index based on transition px divided by size of elements,
+         * every element should have same width, otherwise it should be looped through all elements,
+         * which is not good for performance
+         */
+        let index: number = Math.abs(Math.ceil(this._currentTransitionPx / size));
+
+        index = index + (halfApproached ? 1 : 0);
+        /** Checking if transition went out of scope of array */
+        if (this.items.toArray()[index]) {
+            return this.items.toArray()[index];
+        } else {
+            return this.items.last;
+        }
+    }
+
+    /** @hidden */
+    private getIndexOfItem(item: CarouselItemInterface): number {
+        return this.items.toArray().findIndex((_item) => _item === item);
+    }
+
+    /** Getting size of carousel, width for horizontal, height for vertical */
+    private _getSize(item: CarouselItemInterface): number {
+        if (this.config.vertical) {
+            return item.getHeight();
+        } else {
+            return item.getWidth();
+        }
+    }
+
+    /** @hidden */
+    private _handlePan(delta: number): void {
+        const distance: number = delta - this._lastDistance;
+
+        this._lastDistance = delta;
+
+        this._transitionCarousel(this._currentTransitionPx + distance);
+    }
+
+    /** @hidden */
+    private _handlePanEnd(delta: number): void {
+        this._handlePan(delta);
+
+        const closestItem: CarouselItemInterface = this._getClosest();
+
+        this.goToItem(closestItem, true);
+
+        this.activeChange.emit({
+            item: closestItem,
+            after: delta < 0
+        });
+
+        this.dragStateChange.emit(false);
+        this._lastDistance = 0;
+    }
+
+    /**
+     * Animates the carousel to the currently selected slide.
+     */
+    private _transitionCarousel(transitionPx: number): void {
+        this._currentTransitionPx = transitionPx;
+
+        if (this.config.vertical) {
+            this._elementRef.nativeElement.style.transform = 'translateY(' + this._currentTransitionPx + 'px)';
+        } else {
+            this._elementRef.nativeElement.style.transform = 'translateX(' + this._currentTransitionPx + 'px)';
+        }
+    }
+
+    /** Pam Start handler, removes transition duration, */
+    private _handlePanStart(): void {
+        this._elementRef.nativeElement.style.transitionDuration = '0s';
+        this._prePanTransitionPx = this._currentTransitionPx;
+        this.dragStateChange.emit(true);
+    }
+
+    /** @hidden */
+    private _hammerSetup(): void {
+        this._hammer = new Hammer(this._elementRef.nativeElement);
+
+        this._hammer.get('pan').set({ direction: Hammer.DIRECTION_ALL });
```

Comment:
Could please apply direction depends on your config? for example, `this.config.vertical` should use only `Hammer.DIRECTION_VERTICAL`

Reply:
@JKMarkowski , do you want me to make this change. this code was present from earlier. so please confirm.

Reply:
changed to Hammer.DIRECTION_VERTICAL inside this.config.vertical condition.
fixed.